load-test/src/main/java/gov/cms/ab2d/loadtest/TestRunner.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    @Override
+    public void setupTest(JavaSamplerContext context) {
+        super.setupTest(context);
+
+        String contracts = context.getParameter("contracts");
+        contractArr = contracts.split(",(\\s)*");
+    }
+
+    @Override
+    public void teardownTest(JavaSamplerContext context) {
+
+    }
+
+    class WorkflowWorker extends Thread {
+
+        private final String contractNumber;
+
+        private final CountDownLatch countDownLatch;
+
+        private final SampleResult mainResult;
+
+        WorkflowWorker(String contractNumber, CountDownLatch countDownLatch, SampleResult mainResult) {
+            this.contractNumber = contractNumber;
+            this.countDownLatch = countDownLatch;
+            this.mainResult = mainResult;
+        }
+
+        @Override
+        public void run() {
+            SampleResult contractResult = new SampleResult();
+            addSubResultToSampleResult(mainResult, contractResult);
+            contractResult.sampleStart();
+            contractResult.setThreadName("Contract " + contractNumber);
+
+            SampleResult exportResult = new SampleResult();
+            contractResult.addSubResult(exportResult);
+            exportResult.setSuccessful(false);
+            exportResult.setThreadName("Export for contract " + contractNumber);
+            exportResult.sampleStart();
+
+            try {
+                HttpResponse<String> exportResponse = apiClient.exportByContractRequest(contractNumber);
+
+                exportResult.sampleEnd();
+                exportResult.setResponseCode(String.valueOf(exportResponse.statusCode()));
+                exportResult.setResponseMessage(exportResponse.body());
+
+                if (exportResponse.statusCode() == 202) {
+                    exportResult.setSuccessful(true);
+                } else {
+                    throw new RuntimeException("Encountered error when exporting for contract " + contractNumber);
+                }
+
+                List<String> contentLocationList = exportResponse.headers().map().get("content-location");
+                String url = contentLocationList.get(0);
+
+                int status = 0;
+                HttpResponse<String> statusResponse = null;
+
+                SampleResult statusResult = new SampleResult();
+                contractResult.addSubResult(statusResult);
+                statusResult.setSuccessful(false);
+                statusResult.setThreadName("Status check for contract " + contractNumber);
+                statusResult.sampleStart();
+
+                // Eventually back off, otherwise the server will infinitely send a 429
+                int i = 0;
+                int delay = 50;
+                boolean finishedStatus = false;
+                while (!finishedStatus) {
+                    Thread.sleep(delay);
+                    statusResponse = apiClient.statusRequest(url);
+                    status = statusResponse.statusCode();
```

Comment:
We'd also want to make sure we produce a metric to indicate how each _individual_ status request is taking in addition to the overall job completion metric that you already have.